services/director-v2/src/simcore_service_director_v2/core/dynamic_sidecar_settings.py

```diff
@@ -0,0 +1,337 @@
+    )
+
+    DYNAMIC_SIDECAR_PORT: PortInt = Field(
+        DEFAULT_FASTAPI_PORT,
+        description="port on which the webserver for the dynamic-sidecar is exposed [DEVELOPMENT ONLY]",
+    )
+
+    DYNAMIC_SIDECAR_EXPOSE_PORT: bool = Field(
+        default=False,
+        description="Publishes the service on localhost for debuging and testing [DEVELOPMENT ONLY]"
+        "Can be used to access swagger doc from the host as http://127.0.0.1:30023/dev/doc "
+        "where 30023 is the host published port",
+    )
+
+    PROXY_EXPOSE_PORT: bool = Field(
+        default=False,
+        description="exposes the proxy on localhost for debuging and testing",
+    )
+
+    DYNAMIC_SIDECAR_DOCKER_NODE_RESOURCE_LIMITS_ENABLED: bool = Field(
+        default=False,
+        description=(
+            "Limits concurrent service saves for a docker node. Guarantees "
+            "that no more than X services use a resource together. "
+            "NOTE: A node can end up with all the services from a single study. "
+            "When the study is closed/opened all the services will try to "
+            "upload/download their data. This causes a lot of disk "
+            "and network stress (especially for low power nodes like in AWS). "
+            "Some nodes collapse under load or behave unexpectedly."
+        ),
+    )
+    DYNAMIC_SIDECAR_DOCKER_NODE_CONCURRENT_RESOURCE_SLOTS: PositiveInt = Field(
+        2, description="Amount of slots per resource on a node"
+    )
+    DYNAMIC_SIDECAR_DOCKER_NODE_SAVES_LOCK_TIMEOUT_S: PositiveFloat = Field(
+        10,
+        description=(
+            "Lifetime of the lock. Allows the system to recover a lock "
+            "in case of crash, the lock will expire and result as released."
+        ),
+    )
+
+    @validator("DYNAMIC_SIDECAR_MOUNT_PATH_DEV", pre=True)
+    @classmethod
+    def auto_disable_if_production(cls, v, values):
+        if v and values.get("DYNAMIC_SIDECAR_SC_BOOT_MODE") == BootModeEnum.PRODUCTION:
+            _logger.warning(
+                "In production DYNAMIC_SIDECAR_MOUNT_PATH_DEV cannot be set to %s, enforcing None",
+                v,
+            )
+            return None
+        return v
+
+    @validator("DYNAMIC_SIDECAR_EXPOSE_PORT", pre=True, always=True)
+    @classmethod
+    def auto_enable_if_development(cls, v, values):
+        if (
+            boot_mode := values.get("DYNAMIC_SIDECAR_SC_BOOT_MODE")
+        ) and boot_mode.is_devel_mode():
+            # Can be used to access swagger doc from the host as http://127.0.0.1:30023/dev/doc
+            return True
+        return v
+
+    @validator("DYNAMIC_SIDECAR_IMAGE", pre=True)
+    @classmethod
+    def strip_leading_slashes(cls, v: str) -> str:
+        return v.lstrip("/")
+
+    @validator("DYNAMIC_SIDECAR_LOG_LEVEL")
+    @classmethod
+    def validate_log_level(cls, v: str) -> str:
+        valid_log_levels = {"DEBUG", "INFO", "WARNING", "ERROR"}
```

Comment:
We have `settings_library.utils_logging.MixinLoggingSettings` . Why not add these in the mixin or/add adapt ... ?

Reply:
this code is very old and we never put it in here